From 93e6da09689d64108b6cbe68d2278850756b0a80 Mon Sep 17 00:00:00 2001
From: Yu Qiaowei <cerf.yu@rock-chips.com>
Date: Thu, 21 Jan 2021 15:50:18 +0800
Subject: [PATCH] video/rockchip: rga2: Fix a crash cause by rga timeout.

Because update_patten_buff_mode is not currently used,
it is removed.

Signed-off-by: Yu Qiaowei <cerf.yu@rock-chips.com>
Change-Id: I48c97e75d27d475eee528ed6c838c67fdaa430f8
---
 drivers/video/rockchip/rga2/rga2_drv.c      |  2 ++
 drivers/video/rockchip/rga2/rga2_mmu_info.c | 11 +++++++++++
 2 files changed, 13 insertions(+)

diff --git a/drivers/video/rockchip/rga2/rga2_drv.c b/drivers/video/rockchip/rga2/rga2_drv.c
index 3d7c244ee82f..50297f4d0c8f 100644
--- a/drivers/video/rockchip/rga2/rga2_drv.c
+++ b/drivers/video/rockchip/rga2/rga2_drv.c
@@ -1076,7 +1076,9 @@ static void rga2_del_running_list_timeout(void)
 	while (!list_empty(&rga2_service.running)) {
 		reg = list_entry(rga2_service.running.next, struct rga2_reg,
 				 status_link);
+#if 0
 		kfree(reg->MMU_base);
+#endif
 		if (reg->MMU_len && tbuf) {
 			if (tbuf->back + reg->MMU_len > 2 * tbuf->size)
 				tbuf->back = reg->MMU_len + tbuf->size;
diff --git a/drivers/video/rockchip/rga2/rga2_mmu_info.c b/drivers/video/rockchip/rga2/rga2_mmu_info.c
index cc66df7a64ab..1bf3c00e1b7e 100644
--- a/drivers/video/rockchip/rga2/rga2_mmu_info.c
+++ b/drivers/video/rockchip/rga2/rga2_mmu_info.c
@@ -1128,6 +1128,14 @@ static int rga2_mmu_info_update_palette_table_mode(struct rga2_reg *reg, struct
     return status;
 }
 
+/*
+ * yqw:
+ * This function is currently not sure whether rga2 is used,
+ * because invalidate/clean cache occupies the parameter
+ * reg->MMU_base, so block this function first, and re-implement
+ * this function if necessary.
+ */
+#if 0
 static int rga2_mmu_info_update_patten_buff_mode(struct rga2_reg *reg, struct rga2_req *req)
 {
     int SrcMemSize, CMDMemSize;
@@ -1207,6 +1215,7 @@ static int rga2_mmu_info_update_patten_buff_mode(struct rga2_reg *reg, struct rg
 
     return status;
 }
+#endif
 
 int rga2_set_mmu_info(struct rga2_reg *reg, struct rga2_req *req)
 {
@@ -1230,9 +1239,11 @@ int rga2_set_mmu_info(struct rga2_reg *reg, struct rga2_req *req)
         case update_palette_table_mode :
             ret = rga2_mmu_info_update_palette_table_mode(reg, req);
             break;
+#if 0
         case update_patten_buff_mode :
             ret = rga2_mmu_info_update_patten_buff_mode(reg, req);
             break;
+#endif
         default :
             ret = -1;
             break;
-- 
2.35.3

